### ISSUE TITLE
[java] UselessParentheses: Parentheses in return statement are incorrectly reported as useless

### ISSUE THREAD

**Issue by SUPERCILEX** [pmd/pmd#199]
**Rule Set:**
`unnecessary.html#UselessParentheses`

**Code Sample demonstrating the issue:**

These are incorrectly reported as useless parentheses:
```java
return (mNumber != null ? mNumber.equals(team.mNumber) : team.mNumber == null)
        && (mKey != null ? mKey.equals(team.mKey) : team.mKey == null)
        && (mTemplateKey != null ? mTemplateKey.equals(team.mTemplateKey) : team.mTemplateKey == null)
        && (mName != null ? mName.equals(team.mName) : team.mName == null);
```

Removing them would result in nested ternary operations returning an incorrect boolean.

**Running PMD through:** Gradle

**Comment by jsotuyod** [comment]
@SUPERCILEX thanks for your report! I have confirmed the issue and already have a working fix locally.

I'm still checking if other cases for the rule are suffering from the same problem (I believe they do), to fix all the scenarios together.

**Comment by SUPERCILEX** [comment]
@jsotuyod That would be amazing!!! I've been suppressing PMD on my `equals()` methods for months now when I finally realized you guys had a public issue tracker. Can't wait for the fix!

**Comment by jsotuyod** [comment]
Just submitted #203 for review to address this issue.
